node/src/types/validator_matrix.rs

```diff
@@ -220,7 +220,7 @@ impl ValidatorMatrix {
         None
     }
 
-    fn read_inner(&self) -> RwLockReadGuard<BTreeMap<EraId, EraValidatorWeights>> {
+    pub(crate) fn read_inner(&self) -> RwLockReadGuard<BTreeMap<EraId, EraValidatorWeights>> {
```

Comment:
I strongly think we should avoid doing this - providing a way to return (and hence hold) a lock makes it more likely to cause a deadlock.

Whatever functionality is required, we should ensure the validator matrix can provide it without returning a lock.

Reply:
Fixed in https://github.com/casper-network/casper-node/pull/3615/commits/4bfa012fde7102637b1f53b95605ae265b53774f